### DISCUSSION TITLE
HDRI for magick

### DISCUSSION THREAD

**Discussion by claus-01** [ImageMagick/ImageMagick#2253] [Help] [answered]
Hi,

yesterday I downloaded magick from the IM website & saved it in $HOME/Software. Unfortunately, I discovered that it doesn't have HDRI support ('sudo ./magick --version' resulted in 'Features: Cipher DPC OpenMP(3.1)'). My question: How can I add (if possible) HDRI support to magick, and how do I manage to invoke magick by just typing 'magick' from any directory? I tried to set the $PATH variable, but to no avail.

TIA,

Claus

**Comment by snibgo** [comment]
I suppose you use some version of Unix, and I use Windows so probably can't help, but you should provide more information.

What version of IM? On what platform? Exactly what did you download, from what IM page?

If you want HDRI but don't have it, you need to rebuild IM with HDRI (or download a pre-built version with HDRI).

**Reply by claus-01** [reply]
Hi,

here's the info: I have IM version '6.9.10-23 Q16 x86_64 20190101' installed on Ubuntu MATE 20.04. This was included in the Ubuntu repositories. Since this version doesn't have HDRI support, I additionally downloaded 'magick' from the IM download page at https://imagemagick.org/script/download.php, but this one doesn't seem to have HDRI support, either (see my first post).

Claus

**Reply by fmw42** [reply]
What was the exact version of 'magick' that you downloaded?

What do you get from

`magick -version`

**Reply by claus-01** [reply]
I get ImageMagick 7.0.10-23 Q16 x86_64 2020-07-05. I have to invoke magick with 'sudo ./magick -version'. Merely typing magick -version results in an error message: `Command 'magick' not found, did you mean:  command 'magic' from deb magic (8.2.157+ds.1-1).` `Try: sudo apt install <deb name>`

Claus

**Reply by fmw42** [reply]
Post everything that is returned, please. Yes, magick -version.  If you cannot do that, then something is wrong with your install or permissions.

**Reply by claus-01** [reply]
> Yes, magick -version. If you cannot do that, then something is wrong with your install or permissions.

I downloaded magick to $HOME/Software. This is a binary file. Maybe it helps to install the RPM version using `alien -i ....rpm`

**Reply by claus-01** [reply]
After downloading both RPM's and installing them via 'alien -i', I am not able to run IM. I get the folowing error message:

 `$ identify -version
identify: error while loading shared libraries: libMagickCore-7.Q16HDRI.so.7: cannot open shared object file: No such file or directory`

I searched for `libMagickCore-7.Q16HDRI.so.7` using Synaptic, but couldn't find it. How do I proceed now?

Claus

**Reply by fmw42** [reply]
With IM 7, use 'magick identify'.

**Reply by claus-01** [reply]
I am getting the same error message. I had to execute `$ chmod +x magick` first. Is this ok?

**Reply by fmw42** [reply]
Sorry, but I am not an expert on either Linux or installation of ImageMagick. But you are doing something wrong if you cannot get magick identify to work. You will need more expert help than I can give.

**Reply by claus-01** [reply]
Ok, thanks.

**Comment by fmw42** [comment]
By default IM 6 comes without HDRI. But IM 7 comes with HDRI.

**Comment by claus-01** [comment]
P. S.: Problem solved! ;-) For some reason, although I uninstalled it, I have now `IM 6.9.10-23 Q16 x86_64 20190101` back & am able to open images with an `.hdr` extension.

Claus

**Comment by claus-01** [comment, marked as answer]
P. S.: To get `magick` to work, I had to add the directory to my PATH, like `export PATH="$HOME/Software/ImageMagick:$PATH"`. Now it works - but not permanently. To make this permanent, I added the following code to `.profiles`:

**# set PATH so it includes user's custom ImageMagick bin if it exists
if [ -d "$HOME/Software/ImageMagick" ] ; then
    PATH="$HOME/Software/ImageMagick:$PATH"
fi**

Here, `$HOME/Software/ImageMagick` is the directory where ther binary file `magick` is included. For your installation, you have to modify this accordingly.

Claus
